### ISSUE TITLE
dpctl.tensor.put returns incorrect result 

### ISSUE THREAD

**Issue by vlad-perevezentsev** [IntelPython/dpctl#1360]
The code below works incorrectly in dpctl when `dpctl.tensor.put` needs to rewrite the duplicate index of an element with a different value

```
import dpctl.tensor as dpt

x = dpt.ones((5,))
ind = dpt.asarray([0, 0])

dpt.put(x, ind, [10, 20])
x
> usm_ndarray([10.,  1.,  1.,  1.,  1.])

import numpy
a = numpy.ones((5,))
numpy.put(a, [0, 0], [10, 20])
a
> array([20.,  1.,  1.,  1.,  1.])

```

**Comment by oleksandr-pavlyk** [comment]
`put` parallelizes assignments, and the present input results in race condition.

**Comment by oleksandr-pavlyk** [comment]
gh-1653 adds a docstring note for ``dpt.put`` warning of the race condition, and provides sample code for resolving it for 1D case.

This addition resolves this issue, which we intend to closed in a few days.

**Comment by oleksandr-pavlyk** [comment]
https://github.com/IntelPython/dpctl/pull/1653 adds a docstring note for dpt.put warning of the race condition, and provides sample code for resolving it for 1D case.

This addition resolves this issue, which we intend to closed in a few days.

**Comment by ndgrigorian** [comment]
Closing this issue as resolved now that #1653 is in and #1384 is closed
